fvm/blocks_test.go

```diff
@@ -0,0 +1,100 @@
+package fvm
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	storageMock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func doTest(t *testing.T, f func(*testing.T, *storageMock.Headers, Blocks, flow.Header)) func(*testing.T) {
+	return func(t *testing.T) {
+
+		headers := new(storageMock.Headers)
+		blockFinder := NewBlockFinder(headers)
+		header := unittest.BlockHeaderFixture()
+		header.Height = 10
+
+		f(t, headers, blockFinder, header)
+
+		headers.AssertExpectations(t)
+	}
+}
+
+func Test_BlockFinder_ReturnsHeaderIfSameHeight(t *testing.T) {
+
+	t.Run("returns header is height is the same", doTest(t, func(t *testing.T, headers *storageMock.Headers, blockFinder Blocks, header flow.Header) {
+		heightFrom, err := blockFinder.ByHeightFrom(10, &header)
+		require.NoError(t, err)
+		require.Equal(t, header, *heightFrom)
+	}))
+
+	t.Run("nil header defaults to ByHeight", doTest(t, func(t *testing.T, headers *storageMock.Headers, blockFinder Blocks, header flow.Header) {
+
+		headers.On("ByHeight", uint64(10)).Return(&header, nil)
+
+		heightFrom, err := blockFinder.ByHeightFrom(10, nil)
+		require.NoError(t, err)
+		require.Equal(t, header, *heightFrom)
+	}))
+
+	t.Run("follows blocks chain", doTest(t, func(t *testing.T, headers *storageMock.Headers, blockFinder Blocks, header flow.Header) {
+
+		header0 := unittest.BlockHeaderFixture()
+		header0.Height = 0
+		header1 := unittest.BlockHeaderWithParentFixture(&header0)
+		header1.Height = 1
+		header2 := unittest.BlockHeaderWithParentFixture(&header1)
+		header2.Height = 2
+		header3 := unittest.BlockHeaderWithParentFixture(&header2)
+		header3.Height = 3
+		header4 := unittest.BlockHeaderWithParentFixture(&header3)
+		header4.Height = 4
+		header5 := unittest.BlockHeaderWithParentFixture(&header4)
+		header5.Height = 5
```

Comment:
```suggestion
		header1 := unittest.BlockHeaderWithParentFixture(&header0)
		header2 := unittest.BlockHeaderWithParentFixture(&header1)
		header3 := unittest.BlockHeaderWithParentFixture(&header2)
		header4 := unittest.BlockHeaderWithParentFixture(&header3)
		header5 := unittest.BlockHeaderWithParentFixture(&header4)
```

Should be able to remove these, as it is handled by the fixture. https://github.com/onflow/flow-go/blob/e726e24465c8aa1965f12cdb95cbdb9ea3f01632/utils/unittest/fixtures.go#L245